ansible/library/fabric_info.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python
+
+DOCUMENTATION = '''
+module: fabric_info.py
+Ansible_version_added:  2.0.0.2
+short_description:   Find SONiC Fabric ASIC inforamtion if applicable for the DUT
+Description:
+        When the testbed has Fabric ASICs, this module helps to collect that information
+        which helps in generating the minigraph
+    Input:
+        hwsku num_asic
+
+    Return Ansible_facts:
+    fabric_info:  SONiC Fabric ASIC information
+
+'''
+
+EXAMPLES = '''
+    - name: get Fabric ASIC info
+      fabric_info: hwsku='ACS-MSN2700' num_asic=1
+'''
+
+RETURN = '''
+      ansible_facts{
+        fabric_info: [{'asicname': 'ASIC0', 'ip_prefix': '10.1.0.33/32', 'ip6_prefix': 'FC00:1::33/128'},
+                      {'asicname': 'ASIC1', 'ip_prefix': '10.1.0.34/32', 'ip6_prefix': 'FC00:1::34/128'}]
+      }
+'''
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            hwsku=dict(required=True, type='str'),
+            num_asic=dict(type='str', required=False),
+            asics_host_pfx=dict(type='str', required=False),
+            asics_host_pfx6=dict(type='str', required=False)
+        ),
+        supports_check_mode=True
+    )
+    m_args = module.params
+    try:
+        fabric_info = []
+        # num_asic may not be present for fixed systems which have no Fabric ASIC.
+        # Then return empty fabric_info
+        if 'num_asic' not in m_args or int(m_args['num_asic']) <= 1:
```

Comment:
do we need to support fabric with just one asic ?

Reply:
yes, I will change this to m_args['num_fabric_asic'] < 1